### PR TITLE
OPPS-225 Worldpay Idempotency key fix

### PR DESCRIPTION
**Pull request by jherreraa**
# Unit tests
Finished in 0.418613 seconds.

138 tests, 756 assertions, 0 failures, 0 errors, 
0 pendings, 0 omissions, 0 notifications
100% passed

# Remote Tests
Finished in 118.498554 seconds.

117 tests, 500 assertions, 0 failures, 0 errors, 
0 pendings, 0 omissions, 0 notifications
100% passed

### CODE REVIEW

File: lib/active_merchant/billing/gateways/worldpay.rb [hunk 1]
```diff
@@ -960,7 +960,12 @@ def headers(options)
         cookie = options[:cookie] || cookie
         headers['Cookie'] = cookie if cookie
 
-        headers['Idempotency-Key'] = idempotency_key if idempotency_key
+        # Required because Worldpay does not accept duplicate idempotency keys
+        # for different transactions, such as in the case of an authorize => capture flow.
+        if @idempotency_key
+          headers['Idempotency-Key'] = @idempotency_key
+          options[:idempotency_key] = SecureRandom.uuid
```
Comment (Buitragox) [inline lib/active_merchant/billing/gateways/worldpay.rb:967]: I think this line should be changing the instance variable
```suggestion
          @idempotency_key = SecureRandom.uuid
```

File: test/unit/gateways/worldpay_test.rb [hunk 2]
```diff
@@ -223,6 +223,12 @@ def test_payment_type_for_credit_card
     assert_equal payment, :credit
   end
 
+  def test_idempotency_header
+    options = { idempotency_key: 'test123' }
+    headers = @gateway.send(:headers, options)
+    assert_not_equal options[:idempotency_key], headers['Idempotency-Key']
```
Comment (Buitragox) [inline test/unit/gateways/worldpay_test.rb:229]: Probably need to change this test too.

**Review by Buitragox — CHANGES_REQUESTED** [review]
Couple of things to fix. I also think it may be worth it to make a test that checks that it's actually changing the idempotency key between requests. Maybe something similar to the test `test_optional_idempotency_key_header` (that test is also missing an `assert_equal`, it's just comparing strings with `==`).
https://github.com/activemerchant/active_merchant/blob/master/test/unit/gateways/worldpay_test.rb#L1505

File: test/unit/gateways/worldpay_test.rb (hunk 2 above)
Comment (jherreraa) [reply test/unit/gateways/worldpay_test.rb]: fixed!

File: lib/active_merchant/billing/gateways/worldpay.rb (hunk 1 above)
Comment (jherreraa) [reply lib/active_merchant/billing/gateways/worldpay.rb]: fixed!

**Review by Buitragox — APPROVED** [review]
I have retested and everything looks good. Unit tests and remote tests are good.

**Pull request merged**